### PR TITLE
Issue 18, support of influxdb v09

### PR DESCRIPTION
**Pull request by McFoggy**
This is quite a big PR to support influxdb v09.

I tried to do my best to:
- keep APIs fluent & compatible
- test some important things (especially the inlining of measurements)

There are still some TODOs in the PR, but I would like to receive already some feedback.

_Known limitations & TODOs:_

1. ~~rebase on master~~
1. squash into a single commit (unless you want to keep them separated to keep track of steps)
1. ~~NaN & Infinite do not work on v09 (see [influxdb issues](https://github.com/influxdb/influxdb/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+NaN))~~
1. ~~transformers for standard metric naming patterns could be created~~
    - ~~known categories naming patterns~~
    ~~from metric `srv-01.prod.cpu_load` and known categories pattern naming ["server", "type"] we could create~~:
        - ~~tags: `server=srv-01` & `type=prod`~~
        - ~~measurement: cpu_load~~
    - ~~key/value naming patterns~~
    ~~from metric `server.srv-01.type.prod.cpu_load` we could create:~~
        - ~~tags: `server=srv-01` & `type=prod`~~
        - ~~measurement: cpu_load~~
1. ~~default tags addition in Builder/Reporter~~
1. ~~some documentation~~
1. ~~more tests~~

From the current state, I'd like if possible a first review of the PR.

I did not fired the PR for integration yet, but more for a review and sharing.

### CODE REVIEW

**Comment by McFoggy** [comment]
@davidB  I think you can now start reviewing the PR for inclusion. If you want me to squash it, I can.
For NaN & Infinite, as there are [pending problems in influxdb v09](https://github.com/influxdb/influxdb/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+NaN) I left as it is.

**Comment by davidB** [comment]
Hi, I'll try to review the code this week (not lot of time).

Thanks for the PR.

**Comment by McFoggy** [comment]
Do not hesitate to come back to me if necessary.

**Comment by cemo** [comment]
@davidB any chance to review?

**Comment by davidB** [comment]
I installed influxdb 0.9 last WE. I hope to review this week.

**Comment by McFoggy** [comment]
I finally decided to skip `NaN` & `Infinite` values for Gauges.
My opinion is that it is the easiest and safer approach to take until influxdb answer to it in influxdb/influxdb/issues/4874 or influxdb/influxdb/issues/4089.

File: src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    
+    public HttpInfluxdbProtocol() {
+        this(DEFAULT_HOST, DEFAULT_PORT);
+    }
+    
+    public HttpInfluxdbProtocol(String host, int port, String user, String password) {
+        this(host, port, user, password, DEFAULT_DATABASE);
+    }
+    
+    public String getUser() {
```
Comment (davidB) [inline src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java]: getter are useless (boilerplate), because field are final.

**Comment by davidB** [comment]
GG.

I prefer public final to getter, and to have consistant indentation. Else it's seems fine. I didn't test it (may after merge).

File: src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java (hunk 1 above)
Comment (McFoggy) [inline src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java]: It's more a matter of taste of how your API should look like : either you expose public final fields or you expose getters which allows you to refactor your internals, to do calculations, to introduce decoration, interception, ... But is is probably not the place for such a debate.

Comment (davidB) [inline src/main/java/metrics_influxdb/api/protocols/HttpInfluxdbProtocol.java]: I'm aware of this "advantages", and I applied it, but I prefer to have less code and simple API and to change the API if I change the behavior, and then notify user about this change (compilation break). http://dwayneb.free.fr/post/why_i_don_t_continue_to_use_getter_and_setter/

**Pull request merged**